src/panels/config/core/ha-config-section-general.ts

```diff
@@ -294,10 +294,7 @@ class HaConfigSectionGeneral extends LitElement {
     this._country = this.hass.config.country;
     this._language = this.hass.config.language;
     this._elevation = this.hass.config.elevation;
-    this._timeZone =
-      this.hass.config.time_zone ||
-      Intl.DateTimeFormat?.().resolvedOptions?.().timeZone ||
-      "Etc/GMT";
+    this._timeZone = this.hass.config.time_zone || "Etc/GMT";
```

Comment:
I removed the local time zone fallback here because:
1. I don't think it's even possible for the core value to not be an IANA string already?  If it can be falsey, then the type is wrong.
2. If the polyfill is in play, "UTC" will result in an error for the current picker.